src/common/lib/python/common_extension.cc

```diff
@@ -452,12 +452,16 @@ PyTypeObject PyTaskType = {
 };
 
 /* Create a PyTask from a C struct. The resulting PyTask takes ownership of the
- * TaskSpec and will deallocate the TaskSpec in the PyTask destructor. */
-PyObject *PyTask_make(TaskSpec *task_spec, int64_t task_size) {
+ * task reply and will deallocate the task reply in the PyTask destructor. */
+PyObject *PyTask_make(uint8_t *task_reply, int64_t task_reply_size) {
```

Comment:
Note the tradeoff is that I have to introduce another copy to simplify this (but I think it's worth it).